### PR TITLE
feat(backend): implement delete peer

### LINKED ISSUE #1154: Implement `deletePeer` mutation
Currently, the `deletePeer` mutation is a no-op, we should hook up the resolver to call a service method, and be able to delete a peer by id.

### PR DESCRIPTION
**Pull request by mkurapov**
## Changes proposed in this pull request

- Implement `deletePeer` mutation

## Context

- fixes #1154 

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

File: packages/backend/src/graphql/resolvers/peer.ts [hunk 1]
```diff
@@ -129,13 +129,40 @@ export const updatePeer: MutationResolvers<ApolloContext>['updatePeer'] =
...
     args,
     ctx
   ): Promise<ResolversTypes['DeletePeerMutationResponse']> => {
-    // TODO:
-    console.log(ctx) // temporary to pass linting
-    return {}
+    const peerService = await ctx.container.use('peerService')
+    return peerService
+      .delete(args.id)
+      .then((peer: Peer | undefined) =>
```
Comment (mkurapov) [inline packages/backend/src/graphql/resolvers/peer.ts:139]: I'm not sure why the resolvers were written with the `.then/.catch` promise style, but keeping for consistency

File: packages/backend/src/peer/service.ts [hunk 2]
```diff
@@ -283,3 +285,14 @@ async function getPeersPage(
...
+
+async function deletePeer(
+  deps: ServiceDependencies,
+  id: string
+): Promise<Peer | undefined> {
+  return Peer.query(deps.knex)
+    .withGraphFetched('asset')
+    .deleteById(id)
+    .returning('*')
+    .first()
```
Comment (mkurapov) [inline packages/backend/src/peer/service.ts:297]: I like the pattern of returning the deleted item, similar to how it was done for `accessToken` [revokation](https://github.com/interledger/rafiki/pull/1073#discussion_r1101732342)

File: packages/backend/src/graphql/resolvers/peer.ts (hunk 1 above)
Comment (sabineschaller) [reply packages/backend/src/graphql/resolvers/peer.ts:139]: That is a very good question. We would have to ask @wilsonianb

File: packages/backend/src/graphql/resolvers/peer.test.ts [hunk 3]
```diff
@@ -13,14 +13,13 @@ import { initIocContainer } from '../..'
 import { Config } from '../../config/app'
 import { truncateTables } from '../../tests/tableManager'
 import { errorToCode, errorToMessage, PeerError } from '../../peer/errors'
-import { Peer as PeerModel } from '../../peer/model'
+import { Peer, Peer as PeerModel } from '../../peer/model'
```
Comment (sabineschaller) [inline packages/backend/src/graphql/resolvers/peer.test.ts:16]: Why do you need the model twice?

Comment (wilsonianb) [reply packages/backend/src/graphql/resolvers/peer.test.ts:16]: You can't have a peer without being a peer.
There's always at least two.

File: packages/backend/src/graphql/resolvers/peer.ts (hunk 1 above)
Comment (wilsonianb) [reply packages/backend/src/graphql/resolvers/peer.ts:139]: Looks like I changed it in
- https://github.com/interledger/rafiki/pull/929

I'm guessing I was inspired by the conciseness of some other resolver we had.

File: packages/backend/src/graphql/resolvers/peer.ts [hunk 4]
```diff
@@ -129,13 +129,40 @@ export const updatePeer: MutationResolvers<ApolloContext>['updatePeer'] =
...
+      .catch((error) => {
+        ctx.logger.error(
+          {
+            id: args.id,
+            error
+          },
+          'error deleting peer'
+        )
+        return {
+          code: '400',
```
[the remark this answers is not in the document]
Comment (mkurapov) [reply packages/backend/src/graphql/resolvers/peer.ts:161]: Should I change the `updatePeer` error handler as well to use `500`?

[the remark this answers is not in the document]
Comment (wilsonianb) [reply packages/backend/src/graphql/resolvers/peer.ts:161]: :grimacing: yeah

**Pull request merged**